internal/dag/policy_test.go

```diff
@@ -295,3 +295,116 @@ func TestLoadBalancerPolicy(t *testing.T) {
 		})
 	}
 }
+
+func TestHeadersPolicy(t *testing.T) {
+	tests := map[string]struct {
+		hp      *contour_api_v1.HeadersPolicy
+		want    HeadersPolicy
+		wantErr bool
+	}{
+		"no percentage unchanged": {
+			hp: &contour_api_v1.HeadersPolicy{
+				Set: []contour_api_v1.HeaderValue{{
+					Name:  "X-App-Weight",
+					Value: "100",
+				}},
+			},
+			want: HeadersPolicy{
+				Set: map[string]string{
+					"X-App-Weight": "100",
+				},
+			},
+		},
+		"simple percentage escape": {
+			hp: &contour_api_v1.HeadersPolicy{
+				Set: []contour_api_v1.HeaderValue{{
+					Name:  "X-App-Weight",
+					Value: "100%",
+				}},
+			},
+			want: HeadersPolicy{
+				Set: map[string]string{
+					"X-App-Weight": "100%%",
```

Comment:
Should this be escaped like this? Meaning, adding the double `%%`?

Reply:
i think this is correct b/c contour will send the escaped value to Envoy, since we want a literal `%` to show up in the resulting header value we have to escape it (https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_conn_man/headers#custom-request-response-headers)

Reply:
Yes, that's right. The user specifies the header value as "100%", Contour sends it to Envoy as "100%%" and it ends up being sent as "100%".

Without the escaping Envoy would throw a route configuration error as anything with a single % is considered to be a dynamic header and 100% is not valid.

Reply:
gotcha, makes sense, thanks for clarifying for me @sunjayBhatia & @erwbgy